### ISSUE TITLE
Add options to --freeze-backbone

### ISSUE THREAD

**Issue by NicoLivesey** [fizyr/keras-retinanet#963]
Hello everyone,

First thank you for your awesome work it is really a pleasure to work with this repository.
I wonder if this would be useful (or already in the roadmap) to add more options to the --freeze-backbone parameter like allowing to freeze certain parts of the backbone and not the whole backbone like it is the case now ?

If I am not wrong (I am still a beginner) it is very common in transfer learning after having trained the last part (in the case you have a big enough dataset) to want to unfreeze some deeper parts of the backbone to train them with a lower learning rate. Tell me if I got this wrong or if this not apply to retinanet (i don't see why).

Thank you very much for your answer !

**Comment by martinzlocha** [comment]
Well there are several other layers connected to the backbone. I personally haven't seen only partial freezing of the backbone itself.

I'm not a maintainer but also from a design standpoint adding parameters which alters backbone itself would be cumbersome as there are several different designs so I would recommend adding that functionality yourself.

**Comment by hgaiser** [comment]
I agree with @martinzlocha , it's difficult to add this functionality which would work in all combinations. It's a nice suggestion, but probably better if you add the functionality where you need it.

**Comment by NicoLivesey** [comment]
Thank you for your help
